lm_eval/tasks/hendrycks_math.py

```diff
@@ -62,21 +65,76 @@ def doc_to_decontamination_query(self, doc):
     def doc_to_target(self, doc):
         return " " + doc["solution"]
 
-    def construct_requests(self, doc, ctx):
-        return rf.greedy_until(ctx, ["\n"])
-
-    def process_results(self, doc, results):
-        retval = 0
-        indices = [pos for pos, char in enumerate(results[0]) if char == "$"]
+    def parse_description(self, description):
```

Comment:
is it possible to reuse `lm_eval.utils.simple_parse_args_string`?

(the logic is slightly different in that function, so maybe you already tried it and ran into an issue :))